### ISSUE TITLE
编译时报错

### ISSUE THREAD

**Issue by liuqian1990** [preinheimer/xhprof#96]
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c: In function ‘hp_get_function_name’:
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c:972: warning: assignment discards qualifiers from pointer target type
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c:1042: warning: passing argument 1 of ‘hp_get_base_filename’ discards qualifiers from pointer target type
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c:926: note: expected ‘char *’ but argument is of type ‘const char *’
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c: In function ‘hp_execute_internal’:
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c:1741: error: ‘zend_execute_data’ has no member named ‘Ts’
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c: In function ‘hp_compile_file’:
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c:1785: warning: passing argument 1 of ‘hp_get_base_filename’ discards qualifiers from pointer target type
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c:926: note: expected ‘char *’ but argument is of type ‘const char *’
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c: In function ‘hp_begin’:
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c:1853: error: lvalue required as left operand of assignment
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c:1856: warning: assignment from incompatible pointer type
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c:1861: warning: assignment from incompatible pointer type
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c: In function ‘hp_stop’:
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c:1926: error: lvalue required as left operand of assignment
/home/webadm/wwwroot/xhprof_new/extension/xhprof.c:1927: warning: assignment from incompatible pointer type

**Comment by aik099** [comment]
I guess something similar will happen for PHP 7 as well.

Have you tried compiling using original Facebook repo (from where this one is forked): https://github.com/phacility/xhprof ?

**Comment by tperalta82** [comment]
Use tideways extension which is an updated xhprof 
https://github.com/tideways/php-profiler-extension

and check my pull request
https://github.com/preinheimer/xhprof/pull/97

**Comment by aik099** [comment]
Not sure what exactly is changed. For example since this project was originally forked from Facebook version the PHP extension was changed in a ways, that doesn't map to changes made in original Facebook XHProf project.

I bet both tideways, uprofiler and other xhprof forks also have their unique changes, that are then used in their bundled UI.

**Comment by tperalta82** [comment]
exactly, it's basically a fork with more features, but seems to keep the base ones.
the only thing that seems to be changed (From base as well) is the naming structure, instead of XHPROF_ it's TIDEWAYS_
